Integrations/integration-ThreatX.yml

```diff
@@ -0,0 +1,692 @@
+        }
+
+        return_outputs(md, ec, results)
+
+    @logger
+    def whitelist_ip(ip):
+        commands = {
+            'command': 'new_whitelist',
+            'entry': {
+                'ip': ip,
+                'description': 'Added by ThreatX Demisto Integration',
+                'created': int(time.time())
+            }
+        }
+
+        return http_request('/lists', commands)
+
+    @logger
+    def whitelist_ip_command():
+        ip = demisto.args().get('ip', None)
+        results = whitelist_ip(ip)
+
+        md = tableToMarkdown('Whitelist IP',
+                             results,
+                             ['Result'],
+                             removeNull=True)
+
+        ec = {
+            'IP.Address(val.Id == obj.Id)': ip
+        }
+
+        return_outputs(md, ec, results)
+
+    @logger
+    def unwhitelist_ip(ip):
+        commands = {
+            'command': 'delete_whitelist',
+            'ip': ip
+        }
+
+        return http_request('/lists', commands)
+
+    @logger
+    def unwhitelist_ip_command():
+        ip = demisto.args().get('ip', None)
+        results = unwhitelist_ip(ip)
+
+        md = tableToMarkdown('Unwhitelist IP',
+                             results,
+                             ['Result'],
+                             removeNull=True)
+
+        ec = {
+            'IP.Address(val.Id == obj.Id)': ip
+        }
+
+        return_outputs(md, ec, results)
+
+    @logger
+    def get_entities(entity_name, entity_id, entity_ip):
+        commands = {
+            'command': 'list',
+            'query': {}
+        }
+
+        if entity_name is not None:
+            entity_names = entity_name.split(',')
+            my_entity_name = {'codenames': entity_names}
+            commands['query'].update(my_entity_name)
+
+        if entity_id is not None:
+            entity_ids = entity_id.split(',')
+            my_entity_id = {'entity_ids': entity_ids}
+            commands['query'].update(my_entity_id)
+
+        if entity_ip is not None:
+            entity_ips = entity_ip.split(',')
+            my_entity_ip = {'ip_addresses': entity_ips}
+            commands['query'].update(my_entity_ip)
+
+        return http_request('/entities', commands)
+
+    @logger
+    def get_entity_risk(entity_id):
+        commands = {
+            'command': 'risk_changes',
+            'id': entity_id
+        }
+
+        return http_request('/entities', commands)
+
+    @logger
+    def get_entities_command():
+        entity_name = demisto.args().get('entity_name', None)
+        entity_id = demisto.args().get('entity_id', None)
+        entity_ip = demisto.args().get('entity_ip', None)
+        results = get_entities(entity_name, entity_id, entity_ip)
+        md = []
+        dbot_scores = []
+        ip_enrich = []
+        output = []
+
+        for entity in results:
+            # Grab the entity risk so we can set the Dbot score for the Actor IPs
+            e_id = entity['id']
```

Comment:
Need to verify `id` is in `entity`, you can use `.get()`

Reply:
done